account_banking_pain_base/models/account_payment_order.py

```diff
@@ -489,7 +489,15 @@ def generate_party_block(
     def generate_remittance_info_block(self, parent_node, line, gen_args):
         remittance_info = etree.SubElement(
             parent_node, 'RmtInf')
-        if line.communication_type == 'normal':
+        if line.local_instrument == "LSV+":
```

Comment:
You declared `local_instrument` field in module `l10n_ch_pain_direct_debit`. This module here is more generic and you cannot access the field from here. Could you use instead another communication_type for making the difference with LSV+ ?

If possible, that would mean that here you don't change nothing, and in `l10n_ch_pain_direct_debit`, you inherit the method to do :
```
if line.communication_type == "LSV+":
    ... all your processing
super().generate_remittance_info_block(...)
```

Reply:
There is no communication_type for LSV+, only ISR, BVR, ISO and normal. I'm not sure one is corresponding, so should I add one ? And for the rest, yes it should be possible

Reply:
After looking a bit more into it, it will be difficult to do the processing before calling the  `generate_remittance_info_block` method. When in the case of a LSV+, we want to have **Ustrd** and **Strd** tags. But those tags must be present endure the **RmtInf** tag and this one is added in bank-payment. So if I do some pre-processing in `l10n_ch_pain_direct_debit`, I must create the **RmtInf** tag and then call the super `generate_remittance_info_block` that is responsible for now of adding this tag. So it will be present 2 times. I am not sure how to proceed

Reply:
Ok so I found a way but I'm blocked by essentially the same problem as with local_instrument: there are no communication_type corresponding to LSV+ and if I add one, where do I add it ? If I do it in `l10n_ch_pain_direct_debit`, I don't have access to it in `bank-payment` and I don't know if I'm authorized to do it in `bank-payment`

Reply:
Est ce que je peux utiliser la `res.partner.bank` associé à la ligne pour déterminer si c'est un LSV+ ?

Reply:
Finally I used a little "hack", where I check the tag of the parent node given to `generate_remittance_info_block` and if it is a **RmtInf**, I know that we are in the case of a LSV+. This is not optimal as other cases could come up in the future and also have a **RmtInf** tag.

Reply:
> After looking a bit more into it, it will be difficult to do the processing before calling the `generate_remittance_info_block` method. When in the case of a LSV+, we want to have **Ustrd** and **Strd** tags. But those tags must be present under the **RmtInf** tag and this one is added in bank-payment. So if I do some pre-processing in `l10n_ch_pain_direct_debit`, I must create the **RmtInf** tag and then call the super `generate_remittance_info_block` that is responsible for now of adding this tag. So it will be present 2 times. I am not sure how to proceed

Do you  really need pre-processing? Can't you just start with super() and only do post-processing (that could potentially override what the super() has done in case it's LSV+).